src/System.Private.ServiceModel/src/System/ServiceModel/Channels/BodyWriter.cs

```diff
@@ -80,6 +81,10 @@ internal BodyWriter OnCreateBufferedCopy(int maxBufferSize, XmlDictionaryReaderQ
         }
 
         protected abstract void OnWriteBodyContents(XmlDictionaryWriter writer);
+        protected virtual Task OnWriteBodyContentsAsync(XmlDictionaryWriter writer)
+        {
+            throw new NotImplementedException();
```

Comment:
This should be delegated in the default case to the sync OnWriteBodyContents, rather than throwing NotImplemented. 

We will probably need to plumb this async change in through our derived BodyWriters too. 

`BodyWriter` is public, so we need to be sure that people don't get nasty surprises if someone ends up calling this through some other async method. 

(Note: `throw ExceptionUtility.ThrowHelperError(NotImplemented.ByDesign)` rather than throwing a NIE)